### PR TITLE
Update UI to allow Arrow Import

### PR DESCRIPTION
**Pull request by darthtrevino**

### CODE REVIEW

File: javascript/tables/src/readTable.ts [hunk 1]
```diff
@@ -33,8 +33,17 @@ export async function readTable(
 	}
 	const table = await textTable(input, schema)
 	const opts = defaultOptions(options)
+	const isTypedFormat = schema.format === DataFormat.ARROW
+
+	// TODO: should we eventually perform some casting/typing work on arrow tables? They're already strongly typed
```
Comment (natoverse) [inline javascript/tables/src/readTable.ts:38]: Maybe we should generate a codebook by reading the arrow typings, rather than inferring.

**Review by natoverse — APPROVED** [review]

File: javascript/tables/src/readTable.ts (hunk 1 above)
Comment (darthtrevino) [reply javascript/tables/src/readTable.ts:38]: Hmm, maybe - I think that has to happen outside of readTable though. It takes a codebook in, but does not emit one as an artifact

Comment (natoverse) [reply javascript/tables/src/readTable.ts:38]: Yeah, it's in generateCodebook atm https://github.com/microsoft/datashaper/blob/main/javascript/tables/src/generateCodebook.ts

Comment (natoverse) [reply javascript/tables/src/readTable.ts:38]: `readTable` assumes you already have the schema, but on import if arrow is specified we can put alternate logic in `generateCodebook` to extract the types I suppose

**Pull request merged**